app/concepts/Payment/wrapper/use_case/buy.rb

```diff
@@ -9,24 +9,29 @@ def initialize(wrapper)
         end
 
         def call(id,password, ticket_params, seat, cinema_hall_id, movie_id)
-          left = @wrapper.getLeftRepository
-          right = @wrapper.getRightRepository
+          left = @wrapper.left_Repository
+          right = @wrapper.right_Repository
           if validate?(id,password, left, right, seat,cinema_hall_id, movie_id)
-            success(left,ticket_params, cinema_hall_id, seat, movie_id)
+            payment(left,ticket_params, cinema_hall_id, seat, movie_id)
           end
         end
     
     private
 
     def validate?(id,password, left_Repository, right_Repository, seat, cinema_hall_id, movie_id)
-      @generateseats = UseCase::CinemaHalls::GenerateSeats.new(right_Repository).call(cinema_hall_id)
-      !password.blank? && left_Repository.where(id: id, cinema_hall_id: cinema_hall_id, movie_id: movie_id) && seat.in?(@generateseats) && (!left_Repository.exists?(:cinema_hall_id => cinema_hall_id, seat: seat, movie_id: movie_id)) && left_Repository.ticket_available?(cinema_hall_id, movie_id)
+      @available_seats = UseCase::CinemaHalls::GenerateSeats.new(right_Repository).call(cinema_hall_id)
```

Comment:
Do you really need to use here the instance variable (`@`)?

Reply:
Well, I was thinking the same.